src/python/pants/source/wrapped_globs.py

```diff
@@ -222,13 +235,13 @@ def files_calculator():
     return files_calculator
 
   @staticmethod
-  def _is_glob_dir_outside_root(glob, root):
+  def _is_glob_dir_outside_root(glob: str, root: str) -> bool:
     # The assumption is that a correct glob starts with the root,
     # even after normalizing.
     glob_path = os.path.normpath(os.path.join(root, glob))
 
     # Check if the glob path has the correct root.
-    return os.path.commonprefix([root, glob_path]) != root
+    return cast(bool, os.path.commonprefix([root, glob_path]) != root)
```

Comment:
Unclear why this is necessary, but I didn't want to waste too much time figuring it out. Hopefully MyPy will fix this in the future.